### PR TITLE
Build Neuron with +coreneuron, as used by neurodamus CI

### PR DESCRIPTION
**Pull request by ferdonline**

### CODE REVIEW

File: bluebrain/deployment/environments/applications_hpc.yaml [hunk 1]
```diff
@@ -51,7 +51,7 @@ spack:
     - neurodamus-thalamus+coreneuron+caliper%intel^coreneuron+knl
     - neuron%intel+coreneuron+tests
     - neuron%nvhpc+coreneuron+tests ^coreneuron+gpu~shared
-    - neuron%intel+debug
+    - neuron%intel+coreneuron+debug
```
Comment (iomaganaris) [inline bluebrain/deployment/environments/applications_hpc.yaml]: Maybe this needs to be
```suggestion
    - neuron%intel+coreneuron+debug build_type=Debug
```
to match https://bbpgitlab.epfl.ch/hpc/sim/blueconfigs/-/blob/main/.tests_setup.sh#L23?

**Comment by ferdonline** [comment]
Strangely `Concretized neuron%intel+coreneuron+debug` shows as available from upstream.
However in neurodamus CI it is rebuilding 🤔 there must be smtg different. See e,g, https://bbpgitlab.epfl.ch/hpc/sim/blueconfigs/-/jobs/463675#L411

**Review by pramodk — APPROVED** [review]
LGTM

**Comment by iomaganaris** [comment]
I see[in the spack CI ](https://bbpgitlab.epfl.ch/hpc/spack/-/jobs/470037) there is:
```
$ installed-hashes|tee specs.txt
...
/jmvbprr neuron%intel+coreneuron+debug build_type=Debug
...
```
and [in the blueconfigs CI](https://bbpgitlab.epfl.ch/hpc/sim/blueconfigs/-/jobs/471834):
```
Installing neuron-8.2.2a-jmvbprr6kf6m4vkns7die7ucbucmtclt
```
So I think that the needed neuron version will be deployed and will stop being rebuilt in the CI. We can merge this PR now

**Review by iomaganaris — APPROVED** [review]

**Pull request merged**